Source/FieldSolver/WarpXPushFieldsEM.cpp

```diff
@@ -322,6 +322,8 @@ void WarpX::PSATDForwardTransformRho (
     const amrex::Vector<std::unique_ptr<amrex::MultiFab>>& charge_cp,
     const int icomp, const int dcomp)
 {
+    if (charge_fp[0] == nullptr) return;
```

Comment:
Added this to avoid complex logic where this function is called, such as what we had previously:
```cpp
    // Do rho FFTs only if needed
    if (WarpX::update_with_rho || WarpX::current_correction || WarpX::do_dive_cleaning)
    {
        PSATDForwardTransformRho(rho_fp, rho_cp, 0,0); // rho old
        PSATDForwardTransformRho(rho_fp, rho_cp, 1,1); // rho new
    }
```